### PR TITLE
WIP: #3 Add decennial population

### PR DESCRIPTION
**Pull request by mgraber**

### CODE REVIEW

File: factfinder/main.py [hunk 1]
```diff
@@ -142,25 +144,31 @@ def calculate_variable(self, pff_variable, geotype) -> pd.DataFrame:
...
 
     def aggregate_horizontal(self, source, v, geotype) -> pd.DataFrame:
         """
-        this function will aggregate multiple acs_variables into 1 pff_variable
+        this function will aggregate multiple census_variables into 1 pff_variable
         e.g. ["B01001_044","B01001_020"] -> "mdpop65t66"
         """
         # Create Variables
-        E_variables = [i + "E" for i in v.acs_variable]
-        M_variables = [i + "M" for i in v.acs_variable]
```
Comment (SPTKL) [inline factfinder/main.py:161]:
```python
M_variables = [i + "M" for i in v.acs_variable] if source != 'decennial' else []
```

File: factfinder/main.py [hunk 2]
```diff
@@ -142,25 +144,31 @@ def calculate_variable(self, pff_variable, geotype) -> pd.DataFrame:
             df = self.aggregate_horizontal(source, v, "tract")
             df = self.aggregate_vertical(df, from_geotype="tract", to_geotype="NTA")
         if geotype == "cd_fp_500":
-            df = self.aggregate_horizontal(source, v, "block group")
-            df = self.aggregate_vertical(
-                df, from_geotype="block group", to_geotype="cd_fp_500"
-            )
+            if source == "decennial":
```
Comment (mgraber) [inline factfinder/main.py:147]: Still need other fp and park

File: factfinder/main.py (hunk 1 above)
Comment (SPTKL) [reply factfinder/main.py:161]: and in later
```python
df["m"] = (df[M_variables] ** 2).sum(axis=1) ** 0.5 if if source != 'decennial' else np.nan
```

File: factfinder/main.py [hunk 3]
```diff
@@ -175,6 +183,10 @@ def aggregate_horizontal(self, source, v, geotype) -> pd.DataFrame:
             df["acs_geoid"] = df["state"] + df["county"]
         elif geotype == "city":
             df["acs_geoid"] = df["state"] + df["place"]
+        elif geotype == "block":
+            df["acs_geoid"] = (
```
Comment (SPTKL) [inline factfinder/main.py:187]: let's change `acs_geoid` to `census_geoid` for all

File: factfinder/main.py [hunk 4]
```diff
@@ -188,14 +222,24 @@ def aggregate_vertical(self, df, from_geotype, to_geotype):
...
         elif from_geotype == "tract" and to_geotype == "cd":
             return tract_to_cd(df)
+        elif from_geotype == "block group" and to_geotype == "cd_fp_500":
+            return block_group_to_cd_fp500(df)
+        elif from_geotype == "block group" and to_geotype == "cd_fp_100":
+            return block_group_to_cd_fp100(df)
+        elif from_geotype == "block group" and to_geotype == "cd_park_access":
+            return block_group_to_cd_park_access(df)
+        elif from_geotype == "block" and to_geotype == "cd_fp_500":
+            return block_group_to_cd_fp500(df)
```
Comment (SPTKL) [inline factfinder/main.py:234]: block_group_to_cd_fp500 -> block_to_cd_fp500
for the following two elif statements too

**Review by SPTKL — APPROVED** [review]

**Comment by SPTKL** [comment]
#3

**Pull request merged**